src/evaluators/UnaryExpression.js

```diff
@@ -139,8 +139,15 @@ function tryToEvaluateOperationOrLeaveAsAbstract(
       throw error;
     }
   }
-  realm.applyEffects(effects);
   let completion = effects[0];
+  if (completion instanceof PossiblyNormalCompletion) {
+    // in this case one of the branches may complete abruptly, which means that
+    // not all control flow branches join into one flow at this point.
+    // Consequently we have to continue tracking changes until the point where
+    // all the branches come together into one.
+    completion = realm.composeWithSavedCompletion(completion);
+  }
+  realm.applyEffects(effects);
```

Comment:
Comment?

Reply:
Was there, I force pushed the previous commit by mistake.